simplex/config.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package simplex
+
+import (
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/validators"
+	"github.com/ava-labs/avalanchego/utils/logging"
+)
+
+// Config wraps all the parameters needed for a simplex engine
+type Config struct {
+	Ctx SimplexChainContext
+	Log logging.Logger
+
+	// Validators is a map of node IDs to their validator information.
+	// This tells the node about the current membership set, and should be consistent
+	// across all nodes in the subnet.
+	Validators map[ids.NodeID]*validators.GetValidatorOutput
+
+	// SignBLS is the signing function used for this node to sign messages.
+	SignBLS SignFunc
+}
+
+// Context is information about the current execution.
+// [SubnetID] is the ID of the subnet this context exists within.
+// [ChainID] is the ID of the chain this context exists within.
+// [NodeID] is the ID of this node
```

Comment:
I don't think the bracket notation is correct to use here, as these aren't godoc linkable symbols. (A good test for this is to run `godoc` locally and just look at how the docs are shown.)

Reply:
I think we are trying to consistently only use brackets when the syntax is appropriate. There are definitely instances in some (not so) old code where we incorrectly used them... But for new code we should do it right